ddl/index.go

```diff
@@ -325,54 +324,40 @@ func (d *ddl) onDropIndex(t *meta.Meta, job *model.Job) error {
 	return errors.Trace(err)
 }
 
-func (d *ddl) fetchRowColVals(txn kv.Transaction, t table.Table, handles []int64, indexInfo *model.IndexInfo) (
-	[]*indexRecord, error) {
-	// Through handles access to get all row keys.
-	handlesLen := len(handles)
-	rowKeys := make([]kv.Key, 0, handlesLen)
-	for _, h := range handles {
-		rowKey := tablecodec.EncodeRecordKey(t.RecordPrefix(), h)
-		rowKeys = append(rowKeys, rowKey)
-	}
+func (d *ddl) fetchRowColVals(txn kv.Transaction, t table.Table, batchOpInfo *indexBatchOpInfo, seekHandle int64) error {
+	cols := t.Cols()
+	idxInfo := batchOpInfo.tblIndex.Meta()
+	batchOpInfo.idxRecords = batchOpInfo.idxRecords[:0]
```

Comment:
I think this operation should be done in outer function.
This function name doesn't give any hint that idxRecords will be cleared.